GES-PersonalProject/Assets/Scripts/Collect.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Collect : MonoBehaviour
+{
+    public Text countText;
+
+    private AudioSource audioSource;
+    private SpriteRenderer spriteRenderer;
+    private BoxCollider2D boxCollider2D;
+    public static int count = 0;
+
+    private void Start()
+    {
+        audioSource = GetComponent<AudioSource>();
+        spriteRenderer = GetComponent<SpriteRenderer>();
+        boxCollider2D = GetComponent<BoxCollider2D>();
+        SetCountText();
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.gameObject.CompareTag("Player"))
+        {
+            count++;
+            SetCountText();
+            //Debug.Log("Item count: " + count);
+            audioSource.Play();
+            spriteRenderer.enabled = false;
+            boxCollider2D.enabled = false;
+            Destroy(gameObject, audioSource.clip.length);
+        }
+    }
+
+    void SetCountText()
+    {
+        countText.text = "Mushrooms (" + count.ToString() + "/5)"; 
+        //countText.text = "Count: " + count.ToString();
+    }
+}
```

Comment:
Looks good.

Reply:
I really love the organization of this one, great job!